docs/proposals-accepted/202209-receive-tenant-external-labels.md

```diff
@@ -0,0 +1,52 @@
+---
+type: proposal
+title: Allow specifying tenant-specific external labels in Receivers
+status: accepted
+owner: haanhvu
+menu: proposals-accepted
+---
+
+## 1 Related links/tickets
+
+* https://github.com/thanos-io/thanos/issues/5434
+
+## 2 What problem(s) are we trying to solve?
+
+We want to add arbitrary external labels to each tenant in Thanos Receiver.
+
+## 3 How is it going to be helpful?
+
+This new functionality is helpful in many use cases. Two typical ones are:
+* Compact tenant blocks based on tenants’ external labels
+* Query data from tenants which share the same external labels
+
+## 4 Pitfalls of the current solution
+
+Currently, we can only add external labels to Receivers. Each tenant in a Receiver only gets one external label, which is the tenant ID. This is not flexible enough to do complex compact and query works at the tenant level.
```

Comment:
We cannot partition tenant tsdb blocks yet, but this change would enable that.

Reply:
@fpetkovski If we don't persist tenant's external labels in TSDB, can we still partition tenant TSDB blocks?

Reply:
I am a little bit confused. The partition is to partition each TSDB block?

Reply:
The word "partition" might lead us to thinking of storage partitioning. So I decided to use "select" instead.